src/widgets/auto-style/category.js

```diff
@@ -10,15 +12,29 @@ var CategoryAutoStyler = AutoStyler.extend({
   },
 
   _generateCategoryRamp: function (sym) {
-    var ramp;
-    var cats = this.dataviewModel.get('allCategoryNames');
-    ramp = sym + ': ' + this.colors.getColorByCategory('OTHER') + ';';
-    ramp += cats.map(function (c, i) {
-      var color = this.colors.getColorByCategory(c);
-      return '[' + this.dataviewModel.get('column') + '=\'' + cats[i] + '\']{\n' + sym + ': ' + color + ';\n}';
-    }.bind(this)).join('\n');
-    return ramp;
+    var cats = this.dataviewModel.get('data');
+    var column = this.dataviewModel.get('column');
+    var ramp = ['ramp([' + column + ']'];
+
+    ramp.push(
```

Comment:
Minor thing: I'd move the generation of the categories to a variable outside, to avoid having that big ugly block below.